ghost/core/core/shared/config/helpers.js

```diff
@@ -64,6 +66,28 @@ const isPrivacyDisabled = function isPrivacyDisabled(privacyFlag) {
     return this.get('privacy')[privacyFlag] === false;
 };
 
+/** @type {string|null} */
+let processTmpDirPath = null;
+
+/**
+ * Get a tmp dir path for the current process
+ *
+ * @returns {string} - tmp dir path for the current process
+ */
+function getProcessTmpDirPath(config) {
+    // Memoize the computed path to avoid re-computing it on each call - The
+    // value should not change during the lifetime of the process.
+    if (processTmpDirPath === null) {
+        const siteHash = crypto.createHash('md5')
+            .update(config.getSiteUrl())
+            .digest('hex');
+
+        processTmpDirPath = path.join(os.tmpdir(), `ghost_${siteHash}_${process.pid}`);
```

Comment:
Is it worth just simplifying this to use use a randomly generated UUID?

Reply:
Ah yeh ofc, since adding the memoization, its fine to use something random here 👍

Reply:
Updated in https://github.com/TryGhost/Ghost/pull/21857/commits/0e4fde065ad65035d691697ec7d42ed75bd3f4d9